pytorch_lightning/accelerators/gpu.py

```diff
@@ -32,6 +32,13 @@
 
 class GPUAccelerator(Accelerator):
     """Accelerator for GPU devices."""
+    
+    def __init__(self) -> None:
+        if not torch.cuda.is_available():
+            raise RuntimeError("GPU Accelerator used but CUDA isn't available.")
+        if torch.cuda.device_count() == 0:
+            raise RuntimeError("GPU Accelerator used but found no CUDA devices available.")
+                               
```

Comment:
alternatively this could go into `setup_environment` but this would simplify the accelerator_connector responsibilities at trainer initialization to fail faster